### ISSUE TITLE
Pandas 0.23.4: Memory Error on doing pd.merge() with smallish csv files

### ISSUE THREAD

**Issue by pybokeh** [pandas-dev/pandas#24938]
#### Code Sample, a copy-pastable example if possible

See gist of code:
https://nbviewer.jupyter.org/gist/pybokeh/434e290a1af3636a3f84097e3aa45a72

#### Problem description

Get memory error - see gist above

#### Expected Output
Not get memory error

#### Output of ``pd.show_versions()``

<details>

INSTALLED VERSIONS
------------------
commit: None
python: 3.7.0.final.0
python-bits: 32
OS: Windows
OS-release: 10
machine: AMD64
processor: Intel64 Family 6 Model 78 Stepping 3, GenuineIntel
byteorder: little
LC_ALL: None
LANG: None
LOCALE: None.None

pandas: 0.23.4
pytest: None
pip: 19.0
setuptools: 39.0.1
Cython: None
numpy: 1.16.0
scipy: 1.2.0
pyarrow: None
xarray: None
IPython: 6.5.0
sphinx: None
patsy: 0.5.0
dateutil: 2.7.3
pytz: 2018.5
blosc: None
bottleneck: None
tables: None
numexpr: None
feather: None
matplotlib: 3.0.0
openpyxl: 2.5.7
xlrd: 1.1.0
xlwt: None
xlsxwriter: None
lxml: 4.2.5
bs4: 4.6.3
html5lib: 1.0.1
sqlalchemy: 1.2.14
pymysql: None
psycopg2: None
jinja2: 2.10
s3fs: None
fastparquet: None
pandas_gbq: None
pandas_datareader: None

</details>

I have the csv files that I can email to someone or provided in a different manner.

**Comment by WillAyd** [comment]
What is the cardinality of the datasets? Certainly possible to use a lot of memory if the sets are N to N

We typically ask for examples to be self-contained, i.e. not requiring external files. If there's a way to simulate the issue that way would be preferable

**Comment by pybokeh** [comment]
@WillAyd In both csv files, there were only about 80K rows and less than 10 columns and they both took up less than 10MB of memory per df.info() output.  Was running the script in Windows 10.  I ran the script with task manager running to monitor memory usage and noticed that I still have plenty of available memory (over 50%) when the memory error occurred.  So I find it strange to get a memory error.  With that said, I just realized there is a new version of pandas, I've upgraded to 0.24 and I no longer have the problem.
